lib/response.js

```diff
@@ -91,7 +91,9 @@ Response.prototype.end = function (data, encoding, callback) {
   this.emit('finish')
 }
 
-Response.prototype.destroy = function () {}
+Response.prototype.destroy = function (err) {
```

Comment:
destroy can be called without an error, and you should emit only in that case. The previous test was correct.

This is the original implementation (for reference): https://github.com/nodejs/node/blob/f59d4e05a209173406465a26330465413264a443/lib/_http_outgoing.js#L288-L306

Reply:
If destroy is called without an `error`, what do I emit then? How do we terminate the request?
My understanding was that regardless of whether it is called with or without an error, we gotta end the request. and `onEndFailure` handler does this well.

Also, regarding the previous test, are you sure it was correct? I already tried with a breakpoint and control never comes at [this line](https://github.com/fastify/light-my-request/pull/102/files#diff-a561630bb56b82342bc66697aee2ad96efddcbc9d150665abd6fb7ecb7c0ab2fL1576)

Reply:
Please check the behavior of Node.js core. I think it emits `'close'` and closes the response.